runtime/vm/createramclass.cpp

```diff
@@ -2934,21 +2957,41 @@ internalCreateRAMClassFromROMClass(J9VMThread *vmThread, J9ClassLoader *classLoa
 				 */
 				if (J9_ARE_ALL_BITS_SET(options, J9_FINDCLASS_FLAG_ANON)) {
 					module = hostClass->module;
-				} else if ((classLoader != javaVM->systemClassLoader)
-					|| ((LOAD_LOCATION_PATCH_PATH == locationType)
-					|| (LOAD_LOCATION_MODULE == locationType))
-				) {
-					U_32 pkgNameLength = (U_32) packageNameLength(romClass);
-
-					omrthread_monitor_enter(javaVM->classLoaderModuleAndLocationMutex);
-					module = findModuleForPackage(vmThread, classLoader, J9UTF8_DATA(className), pkgNameLength);
-					omrthread_monitor_exit(javaVM->classLoaderModuleAndLocationMutex);
 				} else {
-					module = javaVM->unamedModuleForSystemLoader;
+					bool findModule = false;
+
+					if (classLoader != javaVM->systemClassLoader) {
+						findModule = true;
+					} else if ((LOAD_LOCATION_PATCH_PATH == locationType)
+							|| (LOAD_LOCATION_MODULE == locationType)) {
+						findModule = true;
+					} else {
+						J9UTF8 *superclassName = J9ROMCLASS_SUPERCLASSNAME(romClass);
+
+						if ((NULL != superclassName)
+						&& J9UTF8_LITERAL_EQUALS(J9UTF8_DATA(superclassName), J9UTF8_LENGTH(superclassName), "java/lang/reflect/Proxy")
+						) {
+							/*
+							 * Proxy classes loaded by the system classloader may belong to a dynamically
+							 * created module (which has been granted access to the interfaces it must implement).
+							 */
```

Comment:
is this a bug fix for something beyond the Access.defineClass issue?  If so, please add it to the issue description